cli/src/semgrep/commands/ci.py

```diff
@@ -219,7 +219,16 @@ def ci(
         output_format=output_format,
     )
 
-    state.metrics.configure(metrics, metrics_legacy)
+    # Metrics are always on for CI scans. It doesn't make sense for them to
+    # be off, since CI scans have to connect to the Cloud Platform
+    # (For reviewers: is this explanation below clear?)
+    if metrics == MetricsState.OFF or metrics_legacy == MetricsState.OFF:
+        raise click.BadParameter(
+            "--metrics cannot be turned off for ci scans since ci scans require access to the Cloud Platform"
```

Comment:
```suggestion
            "--metrics cannot be turned off for ci scans since they require access to Semgrep Cloud Platform"
```

Reply:
or "--metrics cannot be turned off because Semgrep Cloud Platform is needed for ci scans"

Reply:
@ievans I took this warning out in my new changes because without the locality it makes it a little harder to understand that check. That being said, it's easy to add a check that logged in `semgrep ci` scans cannot be run with `--metrics=off` if that's what we want. I have no strong feelings here.

Argument for disallowing `semgrep ci --metrics=off` when logged in: let's be extra clear to users that we are sending their data and there's no way around that when they run semgrep ci
Argument for respecting `semgrep ci --metrics=off` (but still sending data to the app): people might be ok with what we send to the app but specifically not want us to send extra metrics. It also makes our tests work 😅 because our tests run `semgrep ci --metrics=off`

The latter would be less work for me but from a product/privacy perspective I don't know which one makes more sense, let me know.